src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -388,9 +389,12 @@ private static Expression CreateRouteHandlerInvocationContextBase(FactoryContext
         // In the event that a constructor matching the arity of the
         // provided parameters is not found, we fall back to using the
         // non-generic implementation of RouteHandlerInvocationContext.
+        Expression paramArray = factoryContext.BoxedArgs.Length > 0
+            ? Expression.NewArrayInit(typeof(object), factoryContext.BoxedArgs)
+            : Expression.Call(ArrayEmptyMethod.MakeGenericMethod(new Type[] { typeof(object) }));
```

Comment:
Shouldn't MakeGenericMethod be hoisted to the stored MethodInfo so we use the same one for every filter factory?